Add support for non-RSA type client host key

malthe opened the pull request:
<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

This adds support for specifying an SSH client host key along with its type, e.g.

```
ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIGTA+qK7VvFL3oZTgbVwFYcp5ZrPiRkPQd8YhVZDH946
```

Previously, only an RSA key was supported.

A key name (which is optional in the _known hosts_ format) is allowed  – but ignored.

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

malthe committed:
Use key name rather than hardcoded string

malthe committed:
Add support for additional client host key types

malthe committed:
Add test for host key with type

malthe committed:
Add key type handling to SFTP hook

uranusjr commented on airflow/providers/ssh/hooks/ssh.py:
```diff
@@ -159,9 +166,15 @@ def __init__(  # pylint: disable=too-many-statements
+                if host_key is not None and self.no_host_key_check is False:
+                    if host_key.startswith("ssh-"):
+                        key_type, host_key = host_key.split(" ")[:2]
+                        key_constructor = _host_key_mappings[key_type[4:]]
```
Not quite sure about the parsing logic here (is the separator always one single space or should we [split with `None`](https://docs.python.org/3/library/stdtypes.html#str.split) instead? should the split be `host_key.split(" ", 2)` instead? etc.), but otherwise LGTM. We can probably just release this first and fix anything if someone complains anyway.

malthe replied:
@uranusjr I have updated to use `None` instead – nice. Also, I have added a slightly more intrusive change which that if you specify a `host_key` then the system will insist on using that key.

This simplifies the documentation as well.

malthe committed:
When specifying a 'host_key', assume that we want to check the key

malthe committed:
Use more general whitespace splitting

malthe committed:
Disable warnings

malthe committed:
Fix name error

malthe committed:
When 'host_key' is provided, always use it

malthe committed:
Fix host key tests

The 'no_host_key_check' setting is no longer looked at when providing a
host key directly.

potiuk reviewed (approved):
Nice!
